src/lib.rs

```diff
@@ -21,13 +21,89 @@ pub struct PanCamPlugin;
 #[derive(Debug, Clone, Copy, SystemSet, PartialEq, Eq, Hash)]
 pub struct PanCamSystemSet;
 
+/// Which keys move the camera in particular directions for keyboard movement
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Reflect)]
+pub struct DirectionKeys {
+    ///  The keys that move the camera up
+    pub up: Vec<KeyCode>,
+    ///  The keys that move the camera down
+    pub down: Vec<KeyCode>,
+    ///  The keys that move the camera left
+    pub left: Vec<KeyCode>,
+    ///  The keys that move the camera right
+    pub right: Vec<KeyCode>,
+}
+
+impl DirectionKeys {
+    /// No keys move the camera
+    pub const NONE: Self = Self {
+        up: vec![],
+        down: vec![],
+        left: vec![],
+        right: vec![],
+    };
+
+    /// The camera is moved by the arrow keys
+    pub fn arrows() -> Self {
+        Self {
+            up: vec![KeyCode::ArrowUp],
+            down: vec![KeyCode::ArrowDown],
+            left: vec![KeyCode::ArrowLeft],
+            right: vec![KeyCode::ArrowRight],
+        }
+    }
+
+    /// The camera is moved by the WASD keys
+    pub fn wasd() -> Self {
+        Self {
+            up: vec![KeyCode::KeyW],
+            down: vec![KeyCode::KeyS],
+            left: vec![KeyCode::KeyA],
+            right: vec![KeyCode::KeyD],
+        }
+    }
+
+    /// The camera is moved by the arrow and WASD keys
+    pub fn arrows_and_wasd() -> Self {
+        Self {
+            up: vec![KeyCode::ArrowUp, KeyCode::KeyW],
+            down: vec![KeyCode::ArrowDown, KeyCode::KeyS],
+            left: vec![KeyCode::ArrowLeft, KeyCode::KeyA],
+            right: vec![KeyCode::ArrowRight, KeyCode::KeyD],
+        }
+    }
+
+    fn direction(&self, keyboard_buttons: &Res<ButtonInput<KeyCode>>) -> Vec2 {
+        let mut direction = Vec2::ZERO;
+
+        if self.left.iter().any(|key| keyboard_buttons.pressed(*key)) {
+            direction.x += 1.;
```

Comment:
Hmm... I think these signs should be flipped. At least that's how my brain expect it to work.

Reply:
I amended the commit, sorry I have no idea how I managed to do that lol.

Reply:
Haha, yeah I was wondering if this was some kind of apple "natural" scrolling thing :P